Source/WeatherContent/WeatherView.swift

```diff
@@ -0,0 +1,76 @@
+//  Copyright © 2016 Gavan Chan. All rights reserved.
+
+import UIKit
+
+final class WeatherView: UIView {
+
+  enum Theme {
+    case blistering
+    case hot
+    case neutral
+    case cold
+
+    fileprivate var gradientColors: (top: UIColor, bottom: UIColor) {
+      switch self {
+      case .blistering:
+        return (.init(rgb: (254, 229, 207)), .init(rgb: (253, 191, 132)))
+      case .hot:
+        return (.init(rgb: (254, 212, 171)), .init(rgb: (255, 240, 233)))
+      case .neutral:
+        return (.init(rgb: (226, 232, 254)), .init(rgb: (255, 249, 255)))
+      case .cold:
+        return (.init(rgb: (255, 249, 255)), .init(rgb: (226, 232, 254)))
+      }
+    }
+  }
+
+  var theme: Theme? {
+
+    didSet {
+
+      let animation: (() -> Void)
+
+      guard let existing: CAGradientLayer = layer as? CAGradientLayer else { fatalError("Class must match `static var layerClass`.") }
+
+      if let theme = theme {
+
+        let colors: (top: UIColor, bottom: UIColor) = theme.gradientColors
+
+        animation = {
+          existing.colors = [colors.top.cgColor, colors.bottom.cgColor]
+          self.backgroundColor = nil
+        }
+
+      } else {
+
+        DispatchQueue.main.async {
```

Comment:
Let's raise an issue regarding this logic branch.

*Both* the gradient and the view's background colour are being set. We can investigate whether it would be possible to set the *just* the gradient layer's colour(s) to `[UIColor.white.cgColor]` as we are doing here and see whether that is sufficient enough for the use case and then be able to remove our dispatching of
```
self.backgroundColor = .white
```
to the main thread.